### PR TITLE
Closes #21.

### LINKED ISSUE #21: pure-upload should allow uploading files after deferred computation when using the manualStart option
Currently, `pure-upload` only allows `manualStart` to start uploading the most recently selected files. In the case that a computation must be deferred when reacting to `onFileSelected`, there is no simple way to manually `start` an upload for those files. I propose that `start` takes an additional `IUploadFile[]` parameter with the files that should be uploaded.

### PR DESCRIPTION
**Pull request by scottis**

### CODE REVIEW

File: package/index.ts [hunk 1]
```diff
@@ -384,10 +384,12 @@ export class UploadArea {
     if (!this.options.manualStart) this.putFilesToQueue();
   }
 
-  private putFilesToQueue(): void {
-    if (!this.fileList) return;
+  private putFilesToQueue(files?: IUploadFile[] | null): void {
```
Comment (keeema) [inline package/index.ts]: `| null` is not needed. Please remove it.

File: package/index.ts [hunk 2]
```diff
@@ -317,9 +317,9 @@ export class UploadArea {
     }
   }
 
-  start(autoClear: boolean = false) {
-    if (this.options.manualStart && this.fileList) {
-      this.putFilesToQueue();
+  start(autoClear: boolean = false, files?: IUploadFile[]) {
+    if (this.options.manualStart && (files || this.fileList)) {
+      this.putFilesToQueue(files);
       if (autoClear) this.clear();
```
Comment (keeema) [inline package/index.ts]: `clear` function has to take files as an argument too.

File: package/index.ts (hunk 1 above)
Comment (scottis) [reply package/index.ts]: Removed.

File: package/index.ts (hunk 2 above)
Comment (scottis) [reply package/index.ts]: Added.

File: package/index.ts [hunk 3]
```diff
@@ -317,15 +317,25 @@ export class UploadArea {
...
   }
 
-  clear() {
-    this.fileList = null;
+  clear(files?: IUploadFile[]) {
+    if (this.fileList && files !== undefined) {
+      for (let i = 0; i < files.length; ++i) {
+        for (let j = 0; j < this.fileList.length; ++j) {
+          if (files[i] === this.fileList[j]) {
+            this.fileList.splice(j--, 1);
```
Comment (keeema) [inline package/index.ts]: Here you change the the list you iterate in. Please, cover this code with test and fix the issue.

Comment (scottis) [reply package/index.ts]: Changed.

I had added `uploadArea.spec.ts` previously. Are there some more tests I should have added?

**Pull request merged**